src/cmd-build-fast

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+set -euo pipefail
+# This command can be run in two different ways.  If COSA_DIR is unset, it
+# must be run from a coreos-assembler working directory.  It will takes new
+# files from overrides/rootfs and applies them
+# on top of a previously built qemu image.  It creates a new deployment.
+# If COSA_DIR is set, it's assumed instead that you are developing on *one*
+# project - it will take the output of `make && make install DESTDIR=` as
+# an overlay on top of the build from COSA_DIR.  Notably, the qcow2 image
+# will end up in the project's working directory - so you can effectively
+# have independent CoreOS builds for each project.
+# This model is likely preferable if you are developing on an individual
+# project - you should turn to the main `cosa build/cosa build-fast` when
+# working on multiple projects.
+
+dn=$(dirname "$0")
+# shellcheck source=src/cmdlib.sh
+. "${dn}"/cmdlib.sh
+
+projectdir=
+projectname=
+if [ -n "${COSA_DIR:-}" ]; then
+    projectdir=$(pwd)
+    projectname=$(basename "${projectdir}")
+    cd "${COSA_DIR}"
+fi
+
+prepare_build
+
+previous_build=$(get_latest_build)
+previous_qemu=
+if [ -z "${previous_build}" ]; then
+    fatal "previous build required for a fast build"
+fi
+previous_builddir=$(get_build_dir "${previous_build}")
+previous_commit=$(jq -r '.["ostree-commit"]' < "${previous_builddir}/meta.json")
+previous_qemu=$(jq -r '.["images"]["qemu"]["path"]' < "${previous_builddir}/meta.json")
+if [ "${previous_qemu}" = "null" ]; then
+    fatal "--shortcut requires a previous build with qemu"
+fi
+echo "Basing on previous build: ${previous_build:-none}"
+
+# Depends https://github.com/ostreedev/ostree/pull/2041/commits/b3bbbd154225e81980546b2c0b5ed98714830696
+version="fastbuild-$(date +"%s")"
+if [ -n "${projectdir}" ]; then
+    cd "${projectdir}"
```

Comment:
This is minor/an aside, though in non-trivial shell scripts like this I find it conceptually easier to just always stay in the same directory and do anything that really needs a `cd` in a subshell.

Reply:
But everything after that is in the project dir?

Reply:
Hmm, it seemed to me like all the paths used were absolute in the `${projectdir}` case.